### PR TITLE
fix: GET devices/list_device_providers endpoint added

### PR DESCRIPTION
**Pull request by matt-the-dev**

### CODE REVIEW

**Comment by matt-the-dev** [comment]
I ran into some environment issues, but the tests worked after uninstalling and reinstalling Ruby with a guide I found

Good future reference for anyone starting up Ruby on a Mac:
https://mac.install.guide/ruby/index.html

I haven't used Ruby before so I might've missed something, but the test runs well and it was hitting the correct endpoint in the logs

File: lib/seam/clients/device_providers.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    class DeviceProviders < BaseClient
+      def list(params = {})
+        request_seam_object(
+          :get,
+          "/devices/list_device_providers",
+          Seam::DeviceProvider,
+          "device_providers",
+          params: params
+        )
+      end
```
Comment (seveibar) [inline lib/seam/clients/device_providers.rb]: the SDK needs to match the api, so if the api is `/devices/list_device_providers` the corresponding ruby sdk call should be `devices.list_device_providers`

**Review by abimaelmartell — DISMISSED** [review]

File: lib/seam/clients/device_providers.rb (hunk 1 above)
Comment (matt-the-dev) [reply lib/seam/clients/device_providers.rb]: should I just move this to the `Devices` class then? `lib/seam/clients/devices.rb`
image

Comment (seveibar) [reply lib/seam/clients/device_providers.rb]: yep!

Comment (matt-the-dev) [reply lib/seam/clients/device_providers.rb]: cool! I moved it all under Devices

**Review by seveibar — APPROVED** [review]

**Comment by seveibar** [comment]
@mc2147 when you merge, make sure to add a prefix

**Comment by seveibar** [comment]
e.g. `fix: ` is probably good for this commit IMO

**Comment by matt-the-dev** [comment]
> e.g. `fix: ` is probably good for this commit IMO

would feat: be better since we're adding a new endpoint?

**Comment by seveibar** [comment]
> > e.g. `fix: ` is probably good for this commit IMO
> 
> would feat: be better since we're adding a new endpoint?

yes but that would make such high numbers http://sentimentalversioning.org/

(either is ok)

**Comment by matt-the-dev** [comment]
> > > e.g. `fix: ` is probably good for this commit IMO
> > 
> > 
> > would feat: be better since we're adding a new endpoint?
> 
> yes but that would make such high numbers http://sentimentalversioning.org/
> 
> (either is ok)

oh right, that would affect our versioning number. I'll go with "fix" then

**Comment by byalextran** [comment]
> I ran into some environment issues, but the tests worked after uninstalling and reinstalling Ruby with a guide I found
> 
> Good future reference for anyone starting up Ruby on a Mac: https://mac.install.guide/ruby/index.html
> 
> I haven't used Ruby before so I might've missed something, but the test runs well and it was hitting the correct endpoint in the logs

at least on a mac, i've found using [ruby-install](https://github.com/postmodern/ruby-install) and [chruby](https://github.com/postmodern/chruby/) extremely helpful for installing and managing ruby versions. if you try to rely on the system ruby for everything, things can get pretty crazy (i.e. incompatible) really fast as different projects require different versions of ruby.

**Comment by matt-the-dev** [comment]
> > I ran into some environment issues, but the tests worked after uninstalling and reinstalling Ruby with a guide I found
> > Good future reference for anyone starting up Ruby on a Mac: https://mac.install.guide/ruby/index.html
> > I haven't used Ruby before so I might've missed something, but the test runs well and it was hitting the correct endpoint in the logs
> 
> at least on a mac, i've found using [ruby-install](https://github.com/postmodern/ruby-install) and [chruby](https://github.com/postmodern/chruby/) extremely helpful for installing and managing ruby versions. if you try to rely on the system ruby for everything, things can get pretty crazy (i.e. incompatible) really fast as different projects require different versions of ruby.

yeah the system ruby was the one giving me trouble. It wasn't recognizing the `rspec` command so I actually followed a guide to overwrite the default Ruby path with the one I installed with `homebrew`. I'll check out those version managers though, thanks Alex!

**Pull request merged**